runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/artifact/BeamFileSystemArtifactStagingService.java

```diff
@@ -165,7 +165,8 @@ public void removeArtifacts(String stagingSessionToken) throws Exception {
     ResourceId artifactsResourceId =
         dir.resolve(ARTIFACTS, StandardResolveOptions.RESOLVE_DIRECTORY);
     LOG.debug("Removing artifacts: {}", artifactsResourceId);
-    FileSystems.delete(Collections.singletonList(artifactsResourceId));
+    FileSystems.delete(
+        Collections.singletonList(artifactsResourceId), StandardMoveOptions.IGNORE_MISSING_FILES);
```

Comment:
Shouldn't this directory be created even in case the job does not stage any files? That would allow us to delete it. If so, missing files would indicate an error which we might not want to swallow.

Reply:
The artifacts base path is only created once there is actually an artifact stored: https://github.com/apache/beam/blob/969bab3341cfe3cfcb99cf559573b5b76542e6ef/runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/artifact/BeamFileSystemArtifactStagingService.java#L210

Reply:
I wonder if we should create this directory if if nothing is staged. Just like the meta files which are also created. Skipping errors on non-existing directories seems like it has potential for bugs.

Reply:
I don't think there is any potential for bugs here. The artifacts are removed before attempting to delete the directory. If there was an issue then it would surface when removing the artifact files.

Reply:
That's true. I still think it would be cleaner to only delete if it is necessary. Just to keep things explicit. Who knows how the delete order might change in the future?

I'd prefer one of these two options:

1. create the directory even if no files were staged, we can then delete it afterwards
2. skip deletion of the directory in case no files were staged, we then do not have to ignore a missing directory

Reply:
Creating the directory unnecessarily doesn't appear as good option to me.

I changed it to make the delete conditional on the existence of artifacts.